alicloud/resource_alicloud_slb_acl.go

```diff
@@ -0,0 +1,316 @@
+	request.AclName = name
+	request.AddressIPVersion = ip_version
+
+	response, err := client.CreateAccessControlList(request)
+	if err != nil {
+		if IsExceptedError(err, SlbAclInvalidActionRegionNotSupport) {
+			return fmt.Errorf("CreateAccessControlList got an error: %#v", err)
+		}
+		if IsExceptedError(err, SlbAclNumberOverLimit) {
+			return fmt.Errorf("CreateAccessControlList got an error: %#v", err)
+		}
+		return fmt.Errorf("CreateAccessControlList got an error: %#v", err)
+	}
+
+	d.SetId(response.AclId)
+
+	if entryList, ok := d.GetOk("entrys"); ok {
+		list := entryList.(*schema.Set).List()
+		if len(list) > 0 {
+			if err := slbAddAccessControlListEntry(client, list, response.AclId); err != nil {
+				return err
+			}
+		}
+	}
+
+	return resourceAlicloudSlbAclRead(d, meta)
+}
+
+func resourceAlicloudSlbAclRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient).slbconn
+
+	request := slb.CreateDescribeAccessControlListAttributeRequest()
+
+	request.AclId = d.Id()
+	acl, err := client.DescribeAccessControlListAttribute(request)
+
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.SetId(acl.AclId)
+	d.Set("name", acl.AclName)
+	d.Set("ip_version", acl.AddressIPVersion)
+
+	if err := d.Set("entrys", flattenSlbAclEntryMappings(acl.AclEntrys.AclEntry)); err != nil {
+		return err
+	}
+
+	if err := d.Set("related_listeners", flattenSlbRelatedListeneryMappings(acl.RelatedListeners.RelatedListener)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAlicloudSlbAclUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*AliyunClient).slbconn
+
+	d.Partial(true)
+
+	if d.HasChange("name") {
+		request := slb.CreateSetAccessControlListAttributeRequest()
+		request.AclId = d.Id()
+		request.AclName = d.Get("name").(string)
+		if _, err := client.SetAccessControlListAttribute(request); err != nil {
+			return fmt.Errorf("SetAccessControlListAttribute set %s  name %s got an error: %#v",
+				d.Id(), request.AclName, err)
+		}
+		d.SetPartial("name")
+	}
+
+	if d.HasChange("entrys") {
+		o, n := d.GetChange("entrys")
+		oe := o.(*schema.Set)
+		ne := n.(*schema.Set)
+		remove := oe.Difference(ne).List()
+		add := ne.Difference(oe).List()
+
+		if len(remove) > 0 {
+			if err := slbRemoveAccessControlListEntry(client, remove, d.Id()); err != nil {
+				return err
+			}
+		}
+
+		if len(add) > 0 {
+			if err := slbAddAccessControlListEntry(client, add, d.Id()); err != nil {
+				return err
+			}
+		}
+
+	}
+
+	d.Partial(false)
+
+	return resourceAlicloudSlbAclRead(d, meta)
+}
+
+func resourceAlicloudSlbAclDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient).slbconn
+
+	request := slb.CreateDeleteAccessControlListRequest()
+	request.AclId = d.Id()
+
+	if _, err := client.DeleteAccessControlList(request); err != nil {
+		return fmt.Errorf("DeleteAccessControlList got an error: %#v", err)
+	}
+
```

Comment:
Please Describe ACL to ensure it has been deleted after running DeleteAccessControlList.

I addition, there are too many concurrences, please use retry to ensure it can be deleted successfully.